### PR TITLE
Fix type mismatching, among other stuff.

### PR DESCRIPTION
**Pull request by Lukas-Batema**
# WIP, do not merge!
- Fix type returns to return the correct type (still not sure why PEP is so lenient or how it can still compile on incorrectly typed, not inferred, types)
- Fix the `requirements.txt` 
- Update `.gitignore` to include files that should be ignored

### CODE REVIEW

File: client/cli_main.py [hunk 1]
```diff
@@ -163,18 +174,18 @@ def main():
...
+                        post_thought(server_url, username, title, enc_mess)
                                      
                         print("Message uploaded successfully!")
                                             
                     elif sub_choice == "3":
                         get_user_friends(server_url)
                         print()
                         
-                        base_64_encr_sym_key = None
+                        base_64_encr_sym_key = str()
```
Comment (shandralor) [inline client/cli_main.py:188]: Not sure this is going to work if you change it to a string already. The problem here is that if you don't enter a friends username for example when you want to read messages it returns and empty variable for the key above or a dictionary. The check is to only continue if the returned value is a string.

Comment (Lukas-Batema) [reply client/cli_main.py:188]: That's why it is an empty string. I'll do some more testing to test the case.

Comment (Lukas-Batema) [reply client/cli_main.py]: This is resolved. It works flawlessly now!

**Comment by Lukas-Batema** [comment]
I'm still working on the return type mismatching though still, it is not ready to merge, as I will remove the header when it is!

**Review by shandralor — APPROVED** [review]

**Comment by Lukas-Batema** [comment]
Some stuff I need to fix from fixing the merge conflict now haha.

**Review by shandralor — APPROVED** [review]
looks good

**Pull request closed without merging**